source/base/timer.cc

```diff
@@ -175,51 +181,48 @@ double Timer::stop ()
       rusage usage;
       getrusage (RUSAGE_SELF, &usage);
       const double dtime = usage.ru_utime.tv_sec + 1.e-6 * usage.ru_utime.tv_usec;
-      cumulative_time += dtime - start_time;
+      last_lap_cpu_time = dtime - start_time;
 
       rusage usage_children;
       getrusage (RUSAGE_CHILDREN, &usage_children);
       const double dtime_children =
         usage_children.ru_utime.tv_sec + 1.e-6 * usage_children.ru_utime.tv_usec;
-      cumulative_time += dtime_children - start_time_children;
+      last_lap_cpu_time += dtime_children - start_time_children;
 
       struct timeval wall_timer;
       gettimeofday(&wall_timer, nullptr);
       last_lap_time = wall_timer.tv_sec + 1.e-6 * wall_timer.tv_usec
                       - start_wall_time;
 #elif defined(DEAL_II_MSVC)
       last_lap_time = windows::wall_clock() - start_wall_time;
-      cumulative_time += windows::cpu_clock() - start_time;
+      last_lap_cpu_time = windows::cpu_clock() - start_time;
 #else
 #  error Unsupported platform. Porting not finished.
 #endif
 
 #ifdef DEAL_II_WITH_MPI
+      this->mpi_data = Utilities::MPI::min_max_avg (last_lap_time,
+                                                    mpi_communicator);
```

Comment:
Previously, `mpi_data` was not used if the `sync_wall_time` flag was not set. In particular,`get_data` and `print_data` didn't return anything useful then. I wanted to fix this and fill this variable in any case. 
`sync_wall_time` just makes sure that `last_lap_time` and `last_lap_cpu_time` (and consequently `cumulative_wall_time` and `cumulative_time`) are set to the maximum value with respect to all processes.
Of course, this change requires a `MPI_Allreduce` call and thus creates overhead also for `sync_wall_time==false`. 
In the end, filling `mpi_data` and `mpi_total_data` with different values on each process in case `sync_wall_time==false` would also be fine with me although it somewhat contradicts the purpose of these variables.